pkg/kubelet/pod_workers.go

```diff
@@ -620,12 +626,26 @@ func (p *podWorkers) UpdatePod(options UpdatePodOptions) {
 	var podUpdates chan podWork
 	var exists bool
 	if podUpdates, exists = p.podUpdates[uid]; !exists {
+		fullname := kubecontainer.GetPodFullName(pod)
+		if p.isConflictedPod(pod) {
+			if pendingWork, exists := p.reservedWorkFullnames[fullname]; exists {
+				pendingPod := pendingWork.Options.Pod
+				if pendingPod.UID != uid {
+					klog.V(4).InfoS("Mark an outdated pod as finished", "pod", klog.KObj(pod), "podUID", pod.UID)
+					p.podSyncStatuses[pendingPod.UID].finished = true
```

Comment:
This isn't safe because there is no guarantee pendingPod.UID doesn't collide with a running pod (and any running pod can't be terminated early).